Ginger/GingerCoreNET/ActionsLib/ActionConversionUtils.cs

```diff
@@ -0,0 +1,192 @@
+                }
+                else
+                {
+                    if (pLocateBy.PropertyType.IsEnum)
+                        pLocateBy.SetValue(newActUIElement, Enum.Parse(pLocateBy.PropertyType, "ByXY"));
+                }
+            }
+            
+            foreach (var item in lst)
+            {
+                if (item != null)
+                {
+                    if (elementLocateBy == eLocateBy.ByXY.ToString())
+                    {
+                        int oldX = 0;
+                        int oldY = 0;
+
+                        string[] oldxy = locateValue.Split(',');
+                        if (oldxy != null && oldxy.Length > 0)
+                        {
+                            int.TryParse(oldxy[0], out oldX);
+                            if (oldxy.Length > 1)
+                            {
+                                int.TryParse(oldxy[1], out oldY);
+                            }
+                        }
+
+                        if (!string.IsNullOrEmpty(locateValue))
+                        {
+                            int x = 0;
+                            int y = 0;
+                            foreach (var prop in item.Properties)
+                            {
+                                if (!string.IsNullOrEmpty(prop.Value) && prop.Name == "X")
+                                {
+                                    int.TryParse(prop.Value, out x);
+                                }
+                                if (!string.IsNullOrEmpty(prop.Value) && prop.Name == "Y")
+                                {
+                                    int.TryParse(prop.Value, out y);
+                                }
+                                if (x > 0 && y > 0)
+                                {
+                                    break;
+                                }
+                            }
+                            if (oldX == x && oldY == y)
+                            {
+                                isPOM = false;
+                                elementInfo = item;
+                                elementInfo.X = x;
+                                elementInfo.Y = y;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        if (elementLocateBy == eLocateBy.ByXPath.ToString() || elementLocateBy == eLocateBy.ByRelXPath.ToString())
+                        {
+                            if (locateValue == item.XPath)
+                            {
+                                elementInfo = item;
+                            }
+                        }
+                        else if (elementLocateBy == eLocateBy.ByName.ToString())
+                        {
+                            if (item.ElementName.Contains(locateValue))
+                            {
+                                elementInfo = item;
+                            }
```

Comment:
1- i don't see why all of this code is needed, you simply need to go over the POM.Locators and see if you have matching locator with same LocateBy and LocateValue as you pulled from ActUIElement erliaer and if yes so change the LocateBy to be POM
2- let's skip X,Y, the chances the X,Y will match is very low

Reply:
Done